native/src/main/java/io/ballerina/stdlib/http/api/service/signature/AllQueryParams.java

```diff
@@ -123,4 +140,19 @@ public void populateFeed(HttpCarbonMessage httpCarbonMessage, ParamHandler param
             }
         }
     }
+
+    private boolean isEnumQueryParamType(Type paramType) {
+        return TypeUtils.getReferredType(paramType).getTag() == UNION_TAG &&
+                ((UnionType) paramType).getMemberTypes()
+                        .stream().allMatch(type -> type.getTag() == TypeTags.FINITE_TYPE_TAG);
+    }
+
+    private Object castEnumQueryParam(UnionType paramType, BString queryValue, String token) {
+        if (paramType.getMemberTypes().stream().anyMatch(memberType ->
+                ((FiniteType) memberType).getValueSpace().contains(queryValue))) {
+            return queryValue;
+        }
+        throw HttpUtil.createHttpError("query param value '" + token + "' does not match enum type",
+                HttpErrorType.QUERY_PARAM_BINDING_ERROR);
```

Comment:
We can just throw an exception here, all the exceptions are converted to a `QueryParamBindingError` at L138

Reply:
Changed in https://github.com/ballerina-platform/module-ballerina-http/pull/1424/commits/ba7f7f96e60cdab1c62c4fb7387ef92c8068aa9c